scripts/windows/setup/IotEdgeSecurityDaemon.ps1

```diff
@@ -1303,13 +1314,17 @@ function Set-ProvisioningMode {
             return $configurationYaml
         }
         else {
-            $selectionRegex = '(?:[^\S\n]*#[^\S\n]*)?provisioning:\s*#?\s*source:\s*".*"\s*#?\s*global_endpoint:\s*".*"\s*#?\s*scope_id:\s*".*"\s*#?\s*registration_id:\s".*"'
+            $selectionRegex = '(?:[^\S\n]*#[^\S\n]*)?provisioning:\s*#?\s*source:\s*".*"\s*#?\s*global_endpoint:\s*".*"\s*#?\s*scope_id:\s*".*"\s*#?\s*registration_id:\s*".*"\s*#?\s*symmetric_key:\s".*"'
             $replacementContent = @(
                 'provisioning:',
                 '  source: ''dps''',
                 '  global_endpoint: ''https://global.azure-devices-provisioning.net''',
                 "  scope_id: '$ScopeId'",
                 "  registration_id: '$RegistrationId'")
+            if ($NULL -ne $SymmetricKey)
```

Comment:
`if ($SymmetricKey -ne '')` or `if ($SymmetricKey)` (PS strings are not nullable, and don't use yoda conditionals.)

Reply:
I get the null check but with yoda, what beef do you have?